src/lib/importSystem/process.ts

```diff
@@ -0,0 +1,526 @@
+import { Model, Op } from 'sequelize';
+import { Readable } from 'stream';
+import { remap, collectChangedValues, lowercaseFirstLetterOfKeys } from '../dataObjectUtils';
+import { filterDataToModel, modelForTable } from '../modelUtils';
+import EncodingConverter from '../stream/encoding';
+import Hasher, { getHash } from '../stream/hasher';
+import S3Client from '../stream/s3';
+import XMLStream, { SchemaNode } from '../stream/xml';
+import ZipStream, { FileInfo as ZipFileInfo } from '../stream/zip';
+import {
+  getNextFileToProcess,
+  recordAvailableDataFiles,
+  setImportFileStatus,
+  updateAvailableDataFileMetadata,
+} from './record';
+import { IMPORT_STATUSES } from '../../constants';
+import db from '../../models';
+import { auditLogger } from '../../logger';
+
+type ProcessDefinition = {
+  fileName: string,
+  encoding: string,
+  tableName: string,
+  keys: string[],
+  remapDef: Record<string, string>;
+};
+
+/**
+ * Process records according to the given process definition and XML client.
+ * @param processDefinition - The process definition object.
+ * @param xmlClient - The XML client object.
+ * @param fileDate -  the data the file was modified
+ * @param recordActions - The record actions object containing arrays of promises for
+ * inserts, updates, and deletes.
+ * @param schema - the name of each of the columns within the data
+ * @returns A promise that resolves to the updated recordActions object and schema.
+ */
+const processRecords = async (
+  processDefinition: ProcessDefinition,
+  xmlClient: XMLStream,
+  fileDate: Date,
+  recordActions: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    inserts: Promise<any>[],
```

Comment:
Are/can we update these any's?

Reply:
I removed all typing as I don't know a better way. Sequelize is all obfuscated classes as return types, So its hard to type things to the result structures returned from their functions